### ISSUE TITLE
This is just an info question.

### ISSUE THREAD

**Issue by upsteer** [jdice/asdb-legacy#1]
What does the currentheading and coarseheading actually determine? I couldn't find a decent answer on the internet so could you just explain it to me roughly by drawing a figure or something? :)

**Comment by bdice** [comment]
Hi @upsteer, thanks for checking out our code! Tell us what you're working on! Are you building an autonomous vehicle?

We used a coordinate system which was aligned with North-South / East-West. Read the TinyGPS library code here: https://github.com/mikalhart/TinyGPS/blob/master/TinyGPS.cpp#L321

The TinyGPS library uses a system where north is 0, east is 90, south is 180, and west is 270.

To understand how to steer the car, we are interested in two directions, which we call "heading" values. The `courseHeading` points from our current GPS location to the next waypoint's GPS location. This is calculated from the TinyGPS `course_to()` function. The `currentHeading` is the direction that the car is actually moving, based on the GPS's readings. This is calculated from the TinyGPS `course()` function. We took a difference between these angles to determine how to steer the car.

For example, if the car is moving to the east (`currentHeading = 90`) and its target location is to the north (`courseHeading = 0`), the car should turn left (using the `getTurnAngle()` function, we would have `directionDifference = -90`).

Essentially, if `courseHeading - currentHeading < 0`, we should turn left, and if `courseHeading - currentHeading > 0`, we should turn right. Reply back if you have more questions!
